testing/integration/install_non_root_test.go

```diff
@@ -0,0 +1,205 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build integration
+
+package integration
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/elastic-agent/internal/pkg/agent/application/paths"
+	"github.com/elastic/elastic-agent/internal/pkg/agent/install"
+	atesting "github.com/elastic/elastic-agent/pkg/testing"
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+)
+
+func TestInstallNonRootWithoutBasePath(t *testing.T) {
```

Comment:
I like these tests but I'm not sure they are covering enough to ensure we never break the non-root mode. I am worried anytime somebody does anything involving a file they will break the non-root mode by failing to set the permissions properly, or placing it in a directory the elastic-agent user can't access by default.

Key things I can think of that don't look like they are covered here:

1. Upgrades.
2. Enrolling with Fleet
3. Shipping monitoring logs to Fleet.
4. Collecting diagnostics.
5. Collecting system logs and metrics as the system integration would do.

Reply:
I wonder if it would be better to run every test in non-root mode by default since that is the hardest to get right, and only run the sanity check tests for the agent as root. So the inverse of this approach since if it works as non-root it is highly likely to work as root.

The endpoint security tests would have to be root at least, that plus an upgrade test as root might be enough as a sanity check.

Reply:
The main difference seems to be the socket path being different and the install operation has to create the user and group, otherwise it seems we are just auditing for incorrect file permissions.

Reply:
I could do that in a follow up PR. Especially because I have to wait for this to be in the SNAPSHOT build before I can write the upgrade test, which has to be in a follow up PR.

Reply:
> I could do that in a follow up PR

Just to confirm, making the non-root agent the default agent in tests? Would be tidier as a follow up PR anyway most likely.

Reply:
I will do a follow up once this change is in the latest snapshot to run as many tests as we can both as root and non-root installations.